### DISCUSSION TITLE
LUA: Storage

### DISCUSSION THREAD

**Discussion by Alex00101** [FrSkyRC/ETHOS-Feedback-Community#3516] [General]
Hi all,
Porting my *TX scripts to Ethos and I have to say there are a lot of new features that I really like and believe they have great potential. 

One of them is the use of storage module - for cases when I need to store some model-specific custom data (wheres in *TX I had to use raw filesystem, or a hack such as custom curve.

Since it looks like I can use storage read/writhe only in widget read/write handlers (otherwise I'm getting storage not read error), is there a way of triggering the write programmatically, i.e. one of the parameters I intend to save is changed via the code?

Is model.dirty the right call for achieving that (especially when called in flight)?

Thanks

**Comment by bsongis-frsky** [comment]
Thanks for the feedback!

Yes you have to call the function `dirty()`

But it won't directly trigger a write on storage, it just marks the data as dirty, so that you will be sure it will be saved properly when you switch to another model or when you shutdown your transmitter.

The write on storage is done in the "main" task, not the "mixer" task, so there is no risk to slow down the calculations inside the mixer.

**Comment by Alex00101** [comment]
Thank you very much, especially for finding time to respond  when you're
probably busy with the new release.
